jobs/dtd-metrics/src/metrics/tokens-trend.metric.ts

```diff
@@ -0,0 +1,46 @@
+import { MetricFactoryFn } from '../services/metrics-producer.service.js'
+import { aggregateTokensCount, getMonthsAgoDate } from '../utils/helpers.utils.js'
+import { TokensTrendMetric } from '../models/metrics.model.js'
+import { TokensStore } from '../services/tokens-store.service.js'
+import { sub } from 'date-fns'
+
+export const getTokensTrendMetric: MetricFactoryFn<'attivitaDellaPiattaforma'> = async () => {
+  const { tokensByDay } = await TokensStore.getInstance()
+
+  function getTimeseriesFromTokensByDay({
+    oldestDate,
+    jump,
+  }: {
+    oldestDate: Date
+    jump: Duration
+  }): Array<{ date: Date; count: number }> {
+    let currentDate = new Date()
+    currentDate.setHours(0, 0, 0, 0)
+    let currentCount: number = aggregateTokensCount(tokensByDay)
+    const timeseriesData: Array<{ date: Date; count: number }> = [{ date: currentDate, count: currentCount }]
+
+    while (oldestDate <= currentDate) {
+      currentDate = sub(currentDate, jump)
+      currentDate.setHours(0, 0, 0, 0)
+      currentCount = aggregateTokensCount(tokensByDay.filter(({ day }) => day <= currentDate))
+      timeseriesData.push({ date: currentDate, count: currentCount })
+    }
+
+    return timeseriesData.reverse()
+  }
+
+  return TokensTrendMetric.parse({
```

Comment:
just curiosity, but is it required here to parse at runtime the result? 
cannot this be enforced at compile time?

Reply:
Yup! You're right. I always tend to runtime type-check everything and this might be getting out of hand :)
In this case the static typecheck can be a lot more useful that the runtime one since we are not operating with "remote" data. Thanks!